test/test_socp.jl

```diff
@@ -1,243 +1,243 @@
     # Solution is approximately x = [1, .93138, 1.04575]
     solve!(p)
-    @fact p.optval --> roughly(1.2717, TOL)
-    @fact evaluate(norm(x, 4.5)) --> roughly(1.2717, TOL)
+    @test isapprox(p.optval, 1.2717, atol=TOL)
+    @test isapprox(evaluate(norm(x, 4.5)), 1.2717, atol=TOL)
   end
 
-  context("rational norm dual norm") do
+  @testset "rational norm dual norm" begin
     v = [0.463339, 0.0216084, -2.07914, 0.99581, 0.889391];
     x = Variable(5);
     q = 1.379;  # q norm constraint that generates many inequalities
     qs = q / (q - 1);  # Conjugate to q
     p = minimize(x' * v);
     p.constraints += (norm(x, q) <= 1);
-    @fact vexity(p) --> ConvexVexity()
+    @test vexity(p) == ConvexVexity()
     solve!(p)  # Solution is -norm(v, q / (q - 1))
-    @fact p.optval --> roughly(-2.144087, TOL)
-    @fact sum(evaluate(x' * v)) --> roughly(-2.144087, TOL)
-    @fact evaluate(norm(x, q)) --> roughly(1, TOL)
-    @fact sum(evaluate(x' * v)) --> roughly(-sum(abs.(v).^qs)^(1/qs), TOL);
+    @test isapprox(p.optval, -2.144087, atol=TOL)
+    @test isapprox(sum(evaluate(x' * v)), -2.144087, atol=TOL)
+    @test isapprox(evaluate(norm(x, q)), 1, atol=TOL)
+    $(Expr(:toplevel, :(@test isapprox(sum(evaluate(x' * v)), -(sum(abs.(v) .^ qs) ^ (1 / qs)), atol=TOL))))
   end
 
-  context("rational norm atom sum") do
+  @testset "rational norm atom sum" begin
     A = [-0.719255  -0.229089;
          -1.33632   -1.37121;
          0.703447  -1.4482];
     b = [-1.82041, -1.67516, -0.866884];
     q = 1.5;
     xvar = Variable(2);
     p = minimize(.5 * sumsquares(xvar) + norm(A * xvar - b, q));
-    @fact vexity(p) --> ConvexVexity();
+    $(Expr(:toplevel, :(@test vexity(p) == ConvexVexity())))
     solve!(p)
     # Compute gradient, check it is zero(ish)
     x_opt = xvar.value;
     margins = A * x_opt - b;
     qs = q / (q - 1);  # Conjugate
     denom = sum(abs.(margins).^q)^(1/qs);
     g = x_opt + A' * (abs.(margins).^(q-1) .* sign.(margins)) / denom;
-    @fact p.optval --> roughly(1.7227, TOL);
-    @fact norm(g, 2)^2 --> roughly(0, TOL);
+    $(Expr(:toplevel, :(@test isapprox(p.optval, 1.7227, atol=TOL))))
+    $(Expr(:toplevel, :(@test isapprox(norm(g, 2) ^ 2, 0, atol=TOL))))
   end
 
-  context("norm consistent with Base") do
+  @testset "norm consistent with Base" begin
     A = randn(4, 4)
     x = Variable(4, 4)
     x.value = A
-    @fact evaluate(norm(x)) --> roughly(norm(A), TOL);
-    @fact evaluate(norm(x, 1)) --> roughly(norm(A, 1), TOL);
-    @fact evaluate(norm(x, 2)) --> roughly(norm(A, 2), TOL);
-    @fact evaluate(norm(x, Inf)) --> roughly(norm(A, Inf), TOL);
-    @fact evaluate(vecnorm(x, 1)) --> roughly(norm(vec(A), 1), TOL);
-    @fact evaluate(vecnorm(x, 2)) --> roughly(norm(vec(A), 2), TOL);
-    @fact evaluate(vecnorm(x, 7)) --> roughly(norm(vec(A), 7), TOL);
-    @fact evaluate(vecnorm(x, Inf)) --> roughly(norm(vec(A), Inf), TOL);
+    $(Expr(:toplevel, :(@test isapprox(evaluate(norm(x)), norm(A), atol=TOL))))
```

Comment:
Why does this need to be wrapped in `toplevel`? Shouldn't `@test isapprox(evaluate(norm(x)), norm(A), atol=TOL)` work?

Reply:
Good catch. This was not intended! The reason is that this was done by a hacky script that proceeds line by line and `parse("foo;")` (note the `;`) gives `Expr(:toplevel, :foo)`.